core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -112,21 +123,26 @@ protected CloseableIterable<FileScanTask> doPlanFiles() {
       Expression filter = shouldIgnoreResiduals() ? Expressions.alwaysTrue() : filter();
       ResidualEvaluator residuals = ResidualEvaluator.unpartitioned(filter);
 
-      return CloseableIterable.withNoopClose(Iterables.transform(table().snapshots(), snap -> {
+      SnapshotEvaluator snapshotEvaluator = new SnapshotEvaluator(filter, MANIFEST_FILE_SCHEMA.asStruct(),
```

Comment:
nit: if you want to keep it on one line, you could just use `MANIFEST_FILE_SCHEMA` in the constructor of the evaluator class (you have access to it there)

Reply:
I might keep it as argument, the class is more re-usable if it takes the schema as an argument, was thinking we could add this for other metadata tables at some point.